src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,60 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getPosts } from '../../api/posts';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
```

Comment:
The `getPosts` function is imported twice from the same module. Consider removing the duplicate import to clean up the code.